invenio_app_ils/ui/src/invenio_app_ils/authentication/services/SessionManager.js

```diff
@@ -0,0 +1,76 @@
+import { authenticationService } from './AuthenticationService';
+
+class SessionManager {
+  constructor() {
+    let tokenData;
+
+    if (process.env.NODE_ENV === 'production') {
+      let encodedToken = authenticationService.getTokenFromDOM();
+      if (encodedToken) {
+        tokenData = authenticationService.decodeProductionToken(encodedToken);
+        this.setUser(encodedToken, tokenData);
+      } else {
+        this.setAnonymous();
+      }
+    } else {
+      if (process.env.REACT_APP_JWT_TOKEN) {
+        tokenData = {
+          sub: process.env.REACT_APP_USER_ID,
+          locationPid: process.env.REACT_APP_LOCATION_ID,
+          roles: [process.env.REACT_APP_USER_ROLE],
+          exp: process.env.REACT_APP_JWT_TOKEN_EXPIRATION,
+          username: process.env.REACT_APP_JWT_USERNAME,
+        };
+        this.setUser(process.env.REACT_APP_JWT_TOKEN, tokenData);
+      } else {
+        this.setAnonymous();
+      }
+    }
+  }
+
+  setAnonymous() {
+    this.user = null;
+    this.token = null;
+    this.authenticated = false;
+  }
+
+  setUser(token, tokenData) {
+    if (tokenData['sub']) {
+      this.user = {
+        id: tokenData['sub'],
+        roles: tokenData['roles'] || [],
+        username: tokenData['username'],
+      };
+      this.token = {
+        value: token,
+        expiration: tokenData['exp'],
+      };
+      this.authenticated = true;
+    } else {
+      this.setAnonymous();
+    }
+  }
+
+  getUser() {
+    return this.user;
```

Comment:
Do these getters make sense, since there is no handling for empty etc ..? Can we just directly expose the user? `SessionManager.user` and `SessionManager.token`